levels/Level02.js

```diff
@@ -0,0 +1,189 @@
+  } else if (hitFromTopOrBottom) {
+    ball.SpeedY *= -1;
+  }
+}
+
+// Level functions and variables
+
+let life = 2;
+let score = 0;
+
+// Ball Variables
+const ball = {
+  height: 10,
+  width: 10,
+  X: canvas.width / 2,
+  Y: canvas.height - 70,
+  SpeedX: 3,
+  SpeedY: -3,
+  Image: ballImage
+};
+ball.X = ball.X - ball.width / 2;
+
+// Paddle Variables
+const paddleP1 = {
+  height: 20,
+  width: 100,
+  X: canvas.width / 2,
+  Y: canvas.height - 70 + 15,
+  SpeedY: 25,
+  Image: paddleImageP1
+};
+paddleP1.X = paddleP1.X - paddleP1.width / 2;
+
+// Obstacle Variables (for testing)
+const obstacle = {
+  height: 100,
+  width: 100,
+  X: canvas.width / 2,
+  Y: canvas.height / 2
+};
+
+obstacle.X = obstacle.X - obstacle.height / 2;
+obstacle.Y = obstacle.Y - obstacle.width / 2;
+
+//Ball Functions
+function drawBall() {
+  ball.X += ball.SpeedX;
+  ball.Y += ball.SpeedY;
+  ctx.drawImage(ballImage, ball.X, ball.Y, ball.width, ball.height);
+  if (detectCollisionWallLeft(ball) || detectCollisionWallRight(ball)) {
+    ball.SpeedX = ball.SpeedX * -1;
+  }
+  if (detectCollisionWallTop(ball)) {
+    ball.SpeedY = ball.SpeedY * -1;
+  }
+  if (detectCollisionBetween(ball, paddleP1)) {
+    ball.SpeedY = ball.SpeedY * -1;
+    ball.SpeedX = ball.SpeedX * 1.15;
+    ball.SpeedY = ball.SpeedY * 1.15;
+    paddleP1.SpeedY = paddleP1.SpeedY * 1.15;
+    score = ++score;
+  }
+  if (detectCollisionWallBottom(ball)) {
+    life = life - 1;
+    ball.X = canvas.width / 2 - ball.width / 2;
+    ball.Y = canvas.height - 70;
+    ball.SpeedY = ball.SpeedY * -1;
+  }
+}
+
+// Paddle Functions
+function drawPaddleP1() {
+  ctx.drawImage(
+    paddleImageP1,
+    paddleP1.X,
+    paddleP1.Y,
+    paddleP1.width,
+    paddleP1.height
+  );
+  if (detectCollisionWallLeft(paddleP1)) {
+    paddleP1.X = 0;
+  }
+  if (detectCollisionWallRight(paddleP1)) {
+    paddleP1.X = canvas.width - paddleP1.width;
+  }
+}
+function movePaddelRightP1() {
+  paddleP1.X += paddleP1.SpeedY;
+}
+function movePaddleLeftP1() {
+  paddleP1.X -= paddleP1.SpeedY;
+}
+//Paddle interaction
+// https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key
+window.addEventListener("keydown", function(e) {
+  if (e.key === "ArrowRight" && paddleP1.X < canvas.width) {
+    movePaddelRightP1();
+  }
+  if (e.key === "ArrowLeft" && paddleP1.X > 0) {
+    movePaddleLeftP1();
+  }
+});
+
+function drawObstacle(object) {
+  ctx.fillRect(object.X, object.Y, object.width, object.height);
+  const collision = detectCollisionBetween(ball, object);
+  if (collision) {
+    consequenceAfterCollisionWithSquare(object);
+  }
+}
+//Game function
+export function drawLevel02() {
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  drawPaddleP1();
+  drawBall();
+  drawObstacle(obstacle);
+  createUserInfo();
+  if (life > 0 && score < 10) {
+    requestAnimationFrame(drawLevel02);
+  } else if (life > 0 && score === 10) {
+    alert(`New Level unlocked!!!`);
+    setNewLevel();
```

Comment:
`setNewLevel();` is not defined

![image](https://user-images.githubusercontent.com/54030139/65898714-79e14100-e3b2-11e9-897f-cdfcfdafee05.png)

You need to import the function like in Level01.js: `import { setNewLevel } from "../app.js";`

Reply:
Until it's not fixed, we can't jump into lvl 3.

Reply:
Thx mate! Should be fixed now. Send you the link! https://1st-try-pong.juliantoscani.now.sh